### ISSUE TITLE
Doesn't Appear to work with Top level node Exports

### ISSUE THREAD

**Issue by dairyisscary** [k-g-a/jest-node-exports-resolver#11]
Hello!

Thanks for the great package! However, I ran into an issue with packages that use a exports for the "top-level" node modules export. This package only appears to work if the import request is for a submodule like `import x from "my-node-package/submodule";`. it is possible for a package to use the `exports` field for root like `import x from "my-node-package";`.

For an example of this, see solid js which uses exports quite extensively to support different code paths in the server vs the browser.

https://github.com/solidjs/solid/blob/main/packages/solid/package.json#L37

In other words, `jest-node-exports-resolver` seems to work correctly when importing `solid-js/web` but not `solid-js`. i was able to work around this by making two changes to the resolver:

```js
    if (!request.startsWith("@")) {
      if (length > 1) {
        packageName = pkgPathParts[0];
        submoduleName = `./${pkgPathParts.slice(1).join("/")}`;
      } else { // <- added this else
        packageName = request;
        submoduleName = ".";
      }
    } else if (length > 2) {
      packageName = pkgPathParts.slice(0, 2).join("/");
      submoduleName = `./${pkgPathParts.slice(2).join("/")}`; // <- I'm not sure if this line works for @namepspace/root packages but i didn't have any packages lie that to test
    } 
```

i also needed to add a check on `conditions` since this change alone broke for the package `expect` (which has a `conditions` of `undefined`) as far as i can tell, this is a good change regardless since we can skip the `resolveExport` function in more cases
```js
        if (typeof exportValue === "string")
          targetFilePath = exportKey.endsWith("*")
            ? exportValue.replace(/\*/, submoduleName.slice(exportKey.length - 1))
            : exportValue;
        else if (conditions && exportValue !== null && typeof exportValue === "object") { // <- Modified this line
          function resolveExport(exportValue, prevKeys) {
```

I have no idea if these changes are good in general. i'd be happy to make a PR if you think this is good idea.

**Comment by piranna** [comment]
> For an example of this, see solid js which uses exports quite extensively to support different code paths in the server vs the browser.
> 
> [solidjs/solid@`main`/packages/solid/package.json#L37](https://github.com/solidjs/solid/blob/main/packages/solid/package.json?rgh-link-date=2022-01-28T11%3A00%3A25Z#L37)

That `package.json` file [is using `node` condition](https://github.com/solidjs/solid/blob/4c08b520a8cb3ed03886fba160154a679dfab8a6/packages/solid/package.json#L47), that's not supported by `jest`. That's the reason why it's not working for you. In fact, [I have another PR open to add support to it](https://github.com/facebook/jest/pull/11924). It's not a problem of `jest-node-exports-resolver`.

**Comment by piranna** [comment]
Do you have a test case that we can replicate?

**Comment by dairyisscary** [comment]
@piranna good idea. i should have just done this from the get go.

https://github.com/dairyisscary/jest-enviorn-root

```shell
$ node --version 
v16.13.0
$ npm install
$ npm test
```
if the test fails, that means that jest is using the server version of `solid-js` (`./dist/server.js`), when i want the `browser` version(`./dist/solid.js`). the test it will pass if it imports the correct version -- i verified that by monkey patching my local installation of `jest-node-exports-resolver`. hope this helps.

**Comment by piranna** [comment]
I've published a new version 1.1.5, can you check if it works?

**Comment by dairyisscary** [comment]
@piranna its working swimmingly! thank you so much for your efforts. I consider this issue closed.

(side note, i was able to pull your 1.1.5 from npm but it looks like you've forgotten to push your changes to github too -- just a heads up)
